### ISSUE TITLE
HW2 questions: edit/delete comment buttons

### ISSUE THREAD

**Issue by jinsun1117** [swsnu/swppfall2019#92]
Regarding the edit/delete comment buttons, per issue #66 , there can be more than one comment on a single article made by a single person. If that is the case, then the page would have to render multiple edit/delete comment buttons(one for each) for a user with many comments(on a single article). In such situation, would wwe have to mark the button tags so that the system can identify which edit comment button goes to which comment? (for example, maybe, <button id='edit-comment-button' comment_id='22'> ......) 
The reason I am asking this question is because,although the HW spec did not explicitly mention any such requirement, it is rather counterintuitive as to how a machine would automatically grade the correct edit/delete function if we do not pass the buttons any information to relate individual buttons with comments. 

조교님 안녕하세요, 수많은 질문에 시간내서 답변주셔서 감사합니다. 

다름이 아니라,이슈 #66 에서 한 글에 한 사람의 댓글이 여러개 달릴수 있다고 하셨는데요, 그렇다면 마찬가지로 edit/delete comment 버튼도 여러개 달려야 할 것 같습니다. 그러나 이때 스펙상으로나 (제 개인적인 생각으로나) 버튼 태그에 '이 버튼은 이 코멘트에 대응하는 것이다'라고 할 수 있는 부분이 없는데, (예를 들자면, 일반적인 버튼들은 id='edit-comment-button'와 같은 것들을 태그에 달고있어 아무리 화면상 배치를 어지럽게 해도 쉽게 해당 버튼을 찾을 수 있지만 버튼과 코멘트를 대응시키는 것은 힘들어 보입니다) 이것이 정상적인 구현인지(여기까지는 과제 해결 차원에서), 만일 그렇다면 이 부분에 대한 채점기는 어떻게 구현이 될지 (이것은 호기심 차원에서) 궁금합니다. 

감사합니다.

**Comment by ktaebum** [comment]
Yes we cannot map button to comment (they are not unique, and we don't have mapping information). So grading on `edit` and `delete` comment will be tested on articles which have a single user comment

---

네 말씀하신 것처럼 버튼과 comment를 대응시킬 수는 없구요 (unique하지 않고 따로 mapping 정보가 없기 때문에), edit delete에 대한 케이스 채점은 단순하게 하나의 댓글만 있는 article에 한정하여 채점이 될 것 같습니다.

**Comment by haesookim** [comment]
조교님 혹시 버튼과 comment를 대응시킬 수 없다는 것이 무슨 의미인지 혹시 설명해 주실 수 있을까요? 그렇다면 스펙 상 여러 개의 댓글/edit&delete 버튼이 있는 article에서는 특정 comment를 수정/삭제할 수 없는 것이 정상인가요?

**Comment by ktaebum** [comment]
@haesookim 
사용자 측면에서는 아무 문제가 없습니다 (저희 눈에는 전체 UI가 다 보이니 버튼과 comment 내용을 대응시킬 수 있으니)
하지만, 이걸 안 보고 selector로만 (혹은 js code로만) 가져온다고 생각하면, `id`를 통해서 `button`을 가져왔을 때, 그 버튼이 어느 comment 내용에 대응 되는지는 알 수 없는 문제가 있습니다.
